service/controller/resource/alerting/alertmanagerconfigsecret/resource.go

```diff
@@ -0,0 +1,177 @@
+	HTTPProxy        string
+	HTTPSProxy       string
+	NoProxy          string
+	OpsgenieKey      string
+	GrafanaAddress   string
+	SlackApiURL      string
+	SlackProjectName string
+	Pipeline         string
+
+	TemplatePath string
+}
+
+type TemplateData struct {
+	Provider         string
+	Installation     string
+	ProxyURL         *string
+	OpsgenieKey      string
+	GrafanaAddress   string
+	SlackApiURL      string
+	SlackProjectName string
+	Pipeline         string
+}
+
+func New(config Config) (*generic.Resource, error) {
+	clientFunc := func(namespace string) generic.Interface {
+		c := config.K8sClient.K8sClient().CoreV1().Secrets(namespace)
+		return wrappedClient{client: c}
+	}
+
+	if config.TemplatePath == "" {
+		config.TemplatePath = path.Join(templateDirectory, templatePath)
+	}
+
+	c := generic.Config{
+		ClientFunc: clientFunc,
+		Logger:     config.Logger,
+		Name:       Name,
+		GetObjectMeta: func(v interface{}) (metav1.ObjectMeta, error) {
+			return getObjectMeta(v, config)
+		},
+		GetDesiredObject: func(v interface{}) (metav1.Object, error) {
+			return toSecret(v, config)
+		},
+		HasChangedFunc: hasChanged,
+	}
+
+	r, err := generic.New(c)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return r, nil
+}
+
+func getObjectMeta(v interface{}, config Config) (metav1.ObjectMeta, error) {
+	cluster, err := key.ToCluster(v)
+	if err != nil {
+		return metav1.ObjectMeta{}, microerror.Mask(err)
+	}
+
+	return metav1.ObjectMeta{
+		Name:      fmt.Sprintf("alertmanager-%s", config.Installation),
+		Namespace: key.Namespace(cluster),
+	}, nil
+}
+
+func toSecret(v interface{}, config Config) (*corev1.Secret, error) {
+	objectMeta, err := getObjectMeta(v, config)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	notificationTemplate, err := ioutil.ReadFile(path.Join(templateDirectory, notificationTemplatePath))
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	alertmanagerConfigSecret, err := toData(v, config)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	secret := &corev1.Secret{
+		ObjectMeta: objectMeta,
+		Data: map[string][]byte{
+			key.AlertmanagerKey():         alertmanagerConfigSecret,
+			"notification-templates.tmpl": notificationTemplate,
+			"opsgenie.key":                []byte(config.OpsgenieKey),
+		},
+		Type: "Opaque",
+	}
+
+	return secret, nil
+}
+
+func toData(v interface{}, config Config) ([]byte, error) {
+	cluster, err := key.ToCluster(v)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	templateData, err := getTemplateData(cluster, config)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	data, err := template.RenderTemplate(templateData, config.TemplatePath)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return data, nil
+}
+
+func getTemplateData(cluster metav1.Object, config Config) (*TemplateData, error) {
+
+	var proxyURL *string = nil
+	if !strings.Contains(config.NoProxy, "api.opsgenie.com") {
```

Comment:
This is a bit of an edge case (i.e. I don't see why would someone want to do this, but) but with this if `NoProxy == "api.opsgenie.com"` we won't include proxy URL in the global config, but we still need it for slack and possibly other receivers.

Personally, I'd just skip the `NoProxy` check as it's more meant to filter out services internal to the cluster and AM only talks to external services. But if you think it would be prudent to have it, we'll need to check all URLs AM might try to talk to.

Reply:
I see a few cases where this could apply. For example, we could proxy to an internal server for messaging but not opsgenie. This might not be a usecase we have now so I think we can remove it here for now